ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/JsonRpcHttpService.java

```diff
@@ -100,6 +101,25 @@ private void validateConfig(final JsonRpcConfiguration config) {
 
     // Handle json rpc requests
     final Router router = Router.router(vertx);
+
+    // Verify Host header to avoid rebind attack.
+    router
+        .route()
+        .handler(
+            event -> {
+              Optional<String> header = Optional.ofNullable(event.request().getHeader("Host"));
+              if (header.isPresent()
+                  && ("*".equals(header.get())
+                      || (config
+                          .hostWhitelist()
+                          .stream()
+                          .anyMatch(s -> s.toLowerCase().equals(header.get().toLowerCase()))))) {
+                event.next();
+              } else {
+                event.response().setStatusCode(403).end();
```

Comment:
I’d move the “if” logic to a separate method to help readability.

Reply:
> Thoughts on how much detail should be provided here

I don’t think it is a problem giving more details in this case. I wonder if we should add a message on the response saying why we blocked the request. This might avoid people getting frustrated because they can’t send requests to the node.

Also, we need to reach out to Pliny to let them know we are adding this new check so they can update the docs.

Reply:
Updated.